src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -105,14 +105,19 @@ public MessageCacheItem getMessages(Iterator<Locale> fallbackLocalesIter) {
     	MessageCacheItem cacheItem = null;
     	if (cacheService.isContainComponent()) { // Item is in cache
     		cacheItem = cacheService.getCacheOfComponent();
-    		if (cacheItem.isExpired()) { // cacheItem has expired
+    		if (cacheItem.getCachedData().isEmpty()) { // This means that the data to be used is from a fallback locale.
+				// If expired, try to first create and store cacheItem for the requested locale in a separate thread.
+				if (cacheItem.isExpired())
+					this.createCacheItemTask(fallbackLocalesIter);
+
+    			// Use cached data of cacheItem.getLocale() --> fallback locale
+				MessagesDTO fallbackLocaleDTO = new MessagesDTO(dto.getComponent(), cacheItem.getLocale(), dto.getProductID(), dto.getVersion());
+				cacheItem = new ComponentService(fallbackLocaleDTO).getMessages(null);
+			}
+			if (cacheItem.isExpired()) {
```

Comment:
Should add an 'else' before if?

Reply:
No, we still want to refresh expired cacheItem even if it is for the fallback locale.

Reply:
This is actually not new. It is the fix for https://github.com/vmware/singleton/issues/664 and https://github.com/vmware/singleton/issues/662

Reply:
I think it's necessary to add it because the cache of fallback locale will be refreshed in line 115 because of the call getMessages with default locale. 
If no 'else', it will be refreshed twice, right?